### PR TITLE
ci: replace unmaintained markdown link checker

### PR DESCRIPTION
**Pull request by ethanndickson**
The old one was flaking a bunch and blocking PRs. This is the one recommended by the maintainer of the old.

### CODE REVIEW

**Comment by ethanndickson** [comment]
* **#15424** <a href="https://app.graphite.dev/github/pr/coder/coder/15424?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @ethanndickson and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: .github/.linkspector.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+dirs:
+  - docs
+ignorePatterns:
+  - pattern: '://localhost'
+  - pattern: '://.*.?example\\.com'
+  - pattern: 'developer.github.com'
+  - pattern: 'docs.github.com'
+  - pattern: 'github.com/<your_github_handle>'
```
Comment (matifali) [inline .github/.linkspector.yml:8]: Let's check for GitHub links, too, to detect if some upstream docs have moved or a repo has been deleted/renamed, etc.
```suggestion
```

File: .github/workflows/weekly-docs.yaml [hunk 2]
```diff
@@ -7,8 +7,8 @@ on:
   pull_request:
     branches:
       - main
-    paths:
-      - "docs/**"
+    # paths:
+    #   - "docs/**"
```
Comment (matifali) [inline .github/workflows/weekly-docs.yaml:11]: do not forget to uncomment before merge :)

File: .github/.linkspector.yml (hunk 1 above)
Comment (ethanndickson) [reply .github/.linkspector.yml:8]: Yeah, I just copied these from the old config, these sound questionable too:
```
  - pattern: 'tailscale.com'
  - pattern: 'wireguard.com'
  ```

File: docs/README.md [hunk 3]
```diff
@@ -19,7 +19,7 @@ You can use:
 
 - Any Web IDE, such as
 
-  - [code-server](https://github.com/coder/code-server)
+  - [code-server](https://broken.link)
```
Comment (matifali) [inline docs/README.md:22]: Same. Do not forget to revert this change.

**Review by matifali — APPROVED** [review]
Thank you for doing this @ethanndickson ❤️

File: .github/.linkspector.yml (hunk 1 above)
Comment (matifali) [reply .github/.linkspector.yml:8]: Yes. Let us remove all, and we can add if we need to.
